Dockerfile

```diff
@@ -25,6 +30,316 @@ RUN pacman -Syu \
   texlive-bin \
   wget \
   --noconfirm
+  
+###############################################################################
+# AUR DEPENDENCIES
+###############################################################################
+
+# we need opam, perl-critic, verilator
+
+# package (sub) dependencies
+RUN pacman -Syu \
+  base-devel \
+  bison \
+  boost \
+  cmake \
+  ocaml \
+  perl-class-inspector \
+  perl-clone \
+  perl-config-tiny \
+  perl-email-address \
+  perl-exception-class \
+  perl-file-remove \
+  perl-file-sharedir \
+  perl-file-which \
+  perl-io-string \
+  perl-list-moreutils \
+  perl-module-build \
+  perl-module-pluggable \
+  perl-module-runtime \
+  perl-params-util \
+  perl-readonly \
+  perl-role-tiny \
+  perl-sub-exporter \
+  perl-test-deep \
+  perl-test-fatal \
+  perl-test-pod \
+  perl-test-pod-coverage \
+  perl-tidy \
+  re2c \
+  scons \
+  --noconfirm
+  
+# create build user and group 
+RUN groupadd -r build
+RUN useradd -r -g build build
+RUN mkdir /tmp/build
+
+# ensure permissions
+RUN chown -R build:build /tmp/build && \
+    chmod -R 771 /tmp/build
+
+# opam and its dependencies, DO NOT CHANGE ORDER! 
+#************************************************
+
+# build opam dependency (clasp)
+USER build 
+WORKDIR /tmp/build
+RUN git clone https://aur.archlinux.org/clasp.git
+WORKDIR /tmp/build/clasp
+RUN makepkg --noconfirm
+USER root 
+WORKDIR /tmp/build/clasp
+RUN pacman --noconfirm -U *.tar.xz
+
+# build opam dependency (gringo)
+USER build 
+WORKDIR /tmp/build
+RUN git clone https://aur.archlinux.org/gringo.git
+WORKDIR /tmp/build/gringo
+RUN makepkg --noconfirm
+USER root 
+WORKDIR /tmp/build/gringo
+RUN pacman --noconfirm -U *.tar.xz
+
+# build opam dependency (aspcud)
+USER build 
+WORKDIR /tmp/build
+RUN git clone https://aur.archlinux.org/aspcud.git
+WORKDIR /tmp/build/aspcud
+RUN makepkg --noconfirm
+USER root 
+WORKDIR /tmp/build/aspcud
+RUN pacman --noconfirm -U *.tar.xz
+
+# build (opam)
+USER build 
+WORKDIR /tmp/build
+RUN git clone https://aur.archlinux.org/opam.git
+WORKDIR /tmp/build/opam
+RUN makepkg --noconfirm
+USER root 
+WORKDIR /tmp/build/opam
+RUN pacman --noconfirm -U *.tar.xz
```

Comment:
It doesn't matter because even yaourt cannot build stuff as root and 
therefore needs to call makepkg with an unprivileged user. It can also not 
manage dependencies very well because of that.


Am 26. Mai 2016 11:34:30 nachm. schrieb Lasse Schuirmann 
<notifications@github.com>:

>> +RUN git clone https://aur.archlinux.org/aspcud.git
>> +WORKDIR /tmp/build/aspcud
>> +RUN makepkg --noconfirm
>> +USER root
>> +WORKDIR /tmp/build/aspcud
>> +RUN pacman --noconfirm -U *.tar.xz
>> +
>> +# build (opam)
>> +USER build
>> +WORKDIR /tmp/build
>> +RUN git clone https://aur.archlinux.org/opam.git
>> +WORKDIR /tmp/build/opam
>> +RUN makepkg --noconfirm
>> +USER root
>> +WORKDIR /tmp/build/opam
>> +RUN pacman --noconfirm -U *.tar.xz
>
> installing yaourt from the antergos repos or archlinuxfr should be two 
> commands...
>
> ---
> You are receiving this because you authored the thread.
> Reply to this email directly or view it on GitHub:
> https://github.com/coala-analyzer/docker-coala-base/pull/5/files/d8276045c50dd143bb45c6d664ababed07a13d50..1ba5b74af8e11af7941b0319a7ca10f182de10dd#r64824957